### ISSUE TITLE
Ultralytics HUB: WARNING ⚠️ Model upload issue. Missing model runs/detect/train/weights/best.pt.

### ISSUE THREAD

**Issue by axelgiu** [ultralytics/hub#299]
### Search before asking

- [X] I have searched the HUB [issues](https://github.com/ultralytics/hub/issues) and found no similar bug report.

### HUB Component

Training

### Bug

Image sizes 1280 train, 1280 val
Using 8 dataloader workers
Logging results to runs/detect/train
Starting training for 600 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
    575/600      24.8G     0.2611     0.1873     0.8586         34       1280: 100%|██████████| 57/57 [00:21<00:00,  2.65it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.62it/s]
                   all        288        594      0.978      0.907      0.953      0.763

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
    576/600      25.1G     0.2557     0.1804     0.8588         52       1280: 100%|██████████| 57/57 [00:19<00:00,  2.98it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.26it/s]
                   all        288        594      0.978      0.907      0.953      0.764

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
    577/600      24.8G     0.2535      0.184     0.8628         35       1280: 100%|██████████| 57/57 [00:19<00:00,  2.99it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.14it/s]
                   all        288        594      0.978      0.907      0.953      0.764

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
    578/600      24.8G     0.2561     0.1829     0.8613         37       1280: 100%|██████████| 57/57 [00:19<00:00,  2.98it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.33it/s]
                   all        288        594      0.978      0.907      0.953      0.764

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
    579/600      25.1G     0.2548     0.1861     0.8662         33       1280: 100%|██████████| 57/57 [00:19<00:00,  2.97it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.27it/s]
                   all        288        594      0.978      0.908      0.953      0.763

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
    580/600      24.8G     0.2443     0.1824     0.8537         52       1280: 100%|██████████| 57/57 [00:19<00:00,  2.97it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.24it/s]
                   all        288        594      0.978      0.908      0.953      0.764

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
    581/600        25G     0.2607      0.189       0.86         46       1280: 100%|██████████| 57/57 [00:19<00:00,  2.97it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.34it/s]
                   all        288        594      0.978      0.908      0.953      0.763

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
    582/600      24.9G     0.2482     0.1793     0.8601         43       1280: 100%|██████████| 57/57 [00:19<00:00,  2.99it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.35it/s]
                   all        288        594      0.978      0.908      0.953      0.764

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
    583/600        25G      0.248     0.1836     0.8611         32       1280: 100%|██████████| 57/57 [00:19<00:00,  2.97it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.33it/s]
                   all        288        594      0.978      0.908      0.953      0.764

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
    584/600      24.8G     0.2557     0.1831     0.8643         46       1280: 100%|██████████| 57/57 [00:19<00:00,  2.98it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.26it/s]
                   all        288        594      0.978      0.908      0.953      0.764

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
    585/600      24.8G     0.2554     0.1835     0.8682         30       1280: 100%|██████████| 57/57 [00:19<00:00,  2.98it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.33it/s]
                   all        288        594      0.978      0.908      0.953      0.764

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
    586/600      24.9G     0.2514     0.1841     0.8583         32       1280: 100%|██████████| 57/57 [00:19<00:00,  2.98it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.29it/s]
                   all        288        594      0.979      0.907      0.953      0.764

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
    587/600      24.8G     0.2517      0.185     0.8645         35       1280: 100%|██████████| 57/57 [00:19<00:00,  2.97it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.35it/s]
                   all        288        594      0.979      0.907      0.953      0.764

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
    588/600      24.9G     0.2487     0.1779     0.8578         32       1280: 100%|██████████| 57/57 [00:19<00:00,  2.99it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.26it/s]
                   all        288        594      0.979      0.907      0.953      0.764

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
    589/600        25G     0.2522      0.179     0.8588         47       1280: 100%|██████████| 57/57 [00:19<00:00,  2.97it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.33it/s]
                   all        288        594      0.979      0.907      0.953      0.763

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
    590/600      24.9G     0.2505     0.1785     0.8606         49       1280: 100%|██████████| 57/57 [00:19<00:00,  2.99it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.33it/s]
                   all        288        594      0.978      0.907      0.953      0.764

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
    591/600      24.8G     0.2463     0.1804     0.8545         42       1280: 100%|██████████| 57/57 [00:19<00:00,  2.98it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.26it/s]
                   all        288        594       0.98      0.907      0.953      0.764

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
    592/600        25G     0.2483     0.1778     0.8574         28       1280: 100%|██████████| 57/57 [00:19<00:00,  2.97it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.24it/s]
                   all        288        594       0.98      0.907      0.953      0.764

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
    593/600      24.8G     0.2414      0.174     0.8552         44       1280: 100%|██████████| 57/57 [00:19<00:00,  2.98it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.28it/s]
                   all        288        594       0.98      0.907      0.953      0.764

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
    594/600      24.8G     0.2485      0.183     0.8592         19       1280: 100%|██████████| 57/57 [00:19<00:00,  2.97it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.28it/s]
                   all        288        594      0.979      0.907      0.953      0.764

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
    595/600      24.8G     0.2476     0.1779     0.8623         36       1280: 100%|██████████| 57/57 [00:19<00:00,  2.97it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.36it/s]
                   all        288        594       0.98      0.907      0.953      0.764

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
    596/600      24.9G     0.2495     0.1818     0.8603         56       1280: 100%|██████████| 57/57 [00:19<00:00,  2.97it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.24it/s]
                   all        288        594       0.98      0.907      0.953      0.764

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
    597/600        25G     0.2496     0.1818     0.8582         38       1280: 100%|██████████| 57/57 [00:19<00:00,  2.96it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.33it/s]
                   all        288        594       0.98      0.907      0.953      0.764

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
    598/600      24.9G     0.2519     0.1825     0.8558         28       1280: 100%|██████████| 57/57 [00:19<00:00,  2.98it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.30it/s]
                   all        288        594      0.978      0.906      0.953      0.764

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
    599/600      24.8G      0.252     0.1857     0.8572         26       1280: 100%|██████████| 57/57 [00:19<00:00,  2.96it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:02<00:00,  2.35it/s]
                   all        288        594       0.98      0.907      0.953      0.764

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
    600/600      24.8G     0.2452     0.1771     0.8537         60       1280: 100%|██████████| 57/57 [00:19<00:00,  2.97it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:06<00:00,  1.05s/it]
                   all        288        594       0.98      0.907      0.953      0.763

26 epochs completed in 0.163 hours.
Optimizer stripped from runs/detect/train/weights/last.pt, 22.7MB
Ultralytics HUB: Syncing final model...
Ultralytics HUB: WARNING ⚠️ Model upload issue. Missing model runs/detect/train/weights/best.pt.
100%|██████████| 48.0/48.0 [00:00<00:00, 27.3kB/s]
Ultralytics HUB: Done ✅
Ultralytics HUB: View model at https://hub.ultralytics.com/models/xxxxx 🚀
The content for this response was already consumed

### Environment

google colab pro+

**Comment by UltralyticsAssistant** [comment]
@axelgiu hello,

Thank you for bringing this issue to our attention. The warning message you're seeing indicates that the `best.pt` weights file, which should be generated during the training process as the best performing model on your validation data, could not be found for upload to Ultralytics HUB.

Here are the steps you can take to troubleshoot this issue:

1. **Check the Weights Directory**: Ensure the `runs/detect/train/weights/` directory exists and that you have the correct permissions to access it. It's possible that the training process might not have been able to save the `best.pt` file due to permission issues or an incorrect path.

2. **Verify Training Completion**: Make sure your model training completed all scheduled epochs without interruptions. If the training was stopped prematurely, the `best.pt` file might not have been saved.

3. **Look for Saving Criteria**: The `best.pt` file is saved when there is an improvement in performance on the validation dataset. If no improvement was observed during training, the file might not have been created. Ensure your model is learning and improving by monitoring the validation loss and accuracy.

4. **Manual Intervention**: If the `best.pt` file was saved but not recognized, you can try to manually locate it in your directory and verify if it’s intact. If you find the file, you can manually upload it to the HUB.

5. **Re-Training**: If none of the above steps work, you may need to retrain your model, ensuring that the save directory is correctly set up and that there are no interruptions during training.

For further details on managing training and understanding the model saving criteria, please refer to the Ultralytics HUB Docs. If after following the suggested steps, you are still encountering the same issue, feel free to provide additional information, and we can assist you further.
